java/memory/memory-core/src/test/java/org/apache/arrow/memory/TestForeignAllocation.java

```diff
@@ -82,26 +84,27 @@ public void wrapForeignAllocationWithAllocationListener() {
     assertEquals(16, listener.getTotalMem());
   }
 
-  @Test(expected = OutOfMemoryException.class)
-  public void wrapForeignAllocationFailedWithAllocationListener() {
-    final long bufferSize = 16;
-    final long limit = bufferSize - 1;
-
-    final CountingAllocationListener listener = new CountingAllocationListener();
-    try (BufferAllocator listenedAllocator =
-        allocator.newChildAllocator("child", listener, 0L, limit)) {
-      UnsafeForeignAllocation allocation = new UnsafeForeignAllocation(bufferSize);
-      try {
-        assertEquals(0, listenedAllocator.getAllocatedMemory());
-        ArrowBuf buf = listenedAllocator.wrapForeignAllocation(allocation);
-        assertEquals(bufferSize, buf.capacity());
-        buf.close();
-        assertTrue(allocation.released);
-      } finally {
-        allocation.release0();
-      }
+    @Test
+    public void wrapForeignAllocationFailedWithAllocationListener() {
+        assertThrows(OutOfMemoryException.class, () -> {
```

Comment:
Please run `mvn spotless:apply` for lint.

Reply:
Awesome, I can do that. I ran checkstyle. I didn't see spotless. I'll make a note of it for future as well!